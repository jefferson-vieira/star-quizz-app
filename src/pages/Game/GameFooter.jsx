import React, { Component } from 'react';

class GameFooter extends Component {
  renderPages = () => {
    const { currentPage, totalItems, getPeople } = this.props;

    const pages = Math.ceil(totalItems / 10) || 0;

    const isActive = page => currentPage === page;

    return Array.from(new Array(pages), (val, index) => {
      const page = index + 1;
      return (
        <li key={page} className={`page-item ${isActive(page) && 'active'}`}>
          <a
            href="#game-header"
            className="page-link"
            title={`Ir para a p치gina ${page}`}
            onClick={() => getPeople(page)}
          >
            {page}
          </a>
        </li>
      );
    });
  };

  render() {
    const { currentPage, getPeople, next, previous } = this.props;

    return (
      <section id="game-footer" className="border-top border-warning pt-3">
        <nav aria-label="Game Navigation">
          <ul className="pagination pagination-lg justify-content-center">
            <li className={`page-item ${!previous && 'disabled'}`}>
              <a
                href="#game-body"
                className="page-link"
                title={`Ir para a p치gina ${currentPage - 1}`}
                onClick={() => getPeople(currentPage - 1)}
              >
                &lArr; Anterior
              </a>
            </li>
            {this.renderPages()}
            <li className={`page-item ${!next && 'disabled'}`}>
              <a
                href="#game-body"
                className="page-link"
                title={`Ir para a p치gina ${currentPage + 1}`}
                onClick={() => getPeople(currentPage + 1)}
              >
                Pr칩ximo &rArr;
              </a>
            </li>
          </ul>
        </nav>
      </section>
    );
  }
}

export default GameFooter;
