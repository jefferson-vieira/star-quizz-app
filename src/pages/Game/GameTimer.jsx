import React, { Component } from 'react';

import Progress from '@/components/Progress';

import { MAX_TIME_IN_SECONDS } from '@/configs';

import timeUtils from '@/utils/time';

class GameTimer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      maxTime: MAX_TIME_IN_SECONDS,
      time: MAX_TIME_IN_SECONDS
    };
  }

  componentDidMount() {
    this.startTimer();
  }

  componentWillUnmount() {
    this.stopGame();
  }

  startTimer = () => {
    const interval = setInterval(() => {
      if (this.state.time <= 0) {
        this.stopGame();
      } else if (!this.props.pauseTimer) {
        this.setState(prevState => ({ time: prevState.time - 1 }));
      }
    }, 1000);
    this.setState({ interval });
  };

  getProgress = () => (this.state.time / this.state.maxTime) * 100;

  stopGame = () => {
    const { interval, maxTime, time } = this.state;

    clearInterval(interval);
    this.props.stopGame(timeUtils.getTime('00:02:00', maxTime - time));
  };

  render() {
    const { maxTime, time } = this.state;

    return (
      <div className="game-header__timer m-3">
        <div className="row align-items-center border border-warning rounded p-3 m-1">
          <div className="col-12 text-center">
            <p id="timer" className="text-border mb-0">
              {timeUtils.getTime('00:02:00', maxTime - time)}
            </p>
          </div>
          <div className="col-12 my-3">
            <Progress valuenow={this.getProgress()} />
          </div>
          <div className="col-12 text-center">
            <button
              type="button"
              className="btn btn-danger"
              title="Desistir"
              onClick={this.stopGame}
            >
              Desistir
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default GameTimer;
