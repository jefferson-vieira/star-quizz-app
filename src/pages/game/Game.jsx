import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Header from './GameHeader';
import Body from './GameBody';

import { operations } from '../../store/ducks/game';

class Game extends Component {
  componentWillMount() {
    this.props.getPeople();
  }

  componentDidCatch() {
    alert('aff');
  }

  render() {
    return (
      <section id="game">
        <Header />
        <Body people={this.props.people} />
      </section>
    );
  }
}

const mapStateToProps = state => ({
  people: state.game.people
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      getPeople: operations.getPeople
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Game);
