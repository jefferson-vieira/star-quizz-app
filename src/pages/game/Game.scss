@import '../../assets/scss/breakpoints';
@import '../../assets/scss/variables';

.game {
  &-header {
    color: $yellow;

    &__timer {
      position: fixed;
      z-index: 10;
      top: 0;

      @include media-breakpoint-up(sm) {
        right: 0;
      }

      @include media-breakpoint-down(sm) {
        left: 1rem;
        right: 1rem;
      }

      & > div {
        background-color: rgba(0, 0, 0, 0.5);
      }

      & p {
        font-size: 2rem;
      }
    }
  }

  &-body {
    display: grid;
    grid-template-columns: 1fr;

    @include media-breakpoint-up(sm) {
      grid-template-columns: 1fr 1fr;
    }

    @include media-breakpoint-up(md) {
      grid-template-columns: 1fr 1fr 1fr;
    }

    @include media-breakpoint-up(lg) {
      grid-template-columns: 1fr 1fr 1fr 1fr;
    }

    @include media-breakpoint-up(xl) {
      grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
    }

    & .card {
      margin: 0.5rem;
    }
  }
}
