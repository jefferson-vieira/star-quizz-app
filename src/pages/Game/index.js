import './Game.scss';
import Game from './Game';

export default Game;
