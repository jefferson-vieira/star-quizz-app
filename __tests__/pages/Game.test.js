import React from 'react';
import { shallow, mount } from 'enzyme';

import Game from '../../src/pages/game/Game';

describe('Testing Game Component...', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Game />);
    expect(wrapper).toMatchSnapshot();
  });
});
