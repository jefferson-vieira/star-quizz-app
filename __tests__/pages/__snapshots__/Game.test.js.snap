// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Game Component... should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Game />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Game",
    },
    "ref": null,
    "rendered": "Game",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Game",
      },
      "ref": null,
      "rendered": "Game",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
